javacord-core/src/main/java/org/javacord/core/audio/AudioConnectionImpl.java

```diff
@@ -307,38 +299,7 @@ public synchronized boolean tryConnect() {
      * @throws InterruptedException If interrupted while waiting.
      */
     public AudioSource getCurrentAudioSourceBlocking(long timeout, TimeUnit unit) throws InterruptedException {
-        AudioSource source;
-        currentSourceLock.lock();
-        try {
-            AtomicBoolean poll = new AtomicBoolean(false);
-            source = currentSource.updateAndGet(currentSource -> {
-                if (currentSource == null) {
-                    // Always poll if the current source is null
-                    poll.set(true);
-                } else {
-                    // If the current source is not null, only poll if it's still queued
-                    poll.set(queue.peek() == currentSource);
-                }
-                return currentSource;
-            });
-            if (poll.get()) {
-                AudioSource sourceToSet = queue.poll(timeout, unit);
-                if (sourceToSet != null) { // If it's null, it timed out
-                    currentSource.set(sourceToSet);
-                    source = sourceToSet;
-                }
-            }
-        } finally {
-            currentSourceLock.unlock();
-        }
-        return source;
-    }
-
-    /**
-     * Removes the current audio source.
-     */
-    public void removeCurrentSource() {
-        currentSource.set(null);
+        return currentSource.get();
```

Comment:
```suggestion
        return currentSource.get(timeout, unit);
```
This completely ignores the `timeout` and `unit` parameter.